P4RT_App_DB_Schema.md

```diff
@@ -0,0 +1,396 @@
+
+; defines the P4RT:ACL_* table entries that correspond to a matching DEFINITION table.
+
+
+|                           |                                                                                               |                                                                                                        |
+|---------------------------|-----------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------|
+| key                     = | P4RT:ACL_<table_name>:<Entry_ID_JSON>                                                         | ; Dynamic table name as used in the DEFINITION entry.                                                  |
+| <Entry_ID_JSON> =         | match/<name>: "<value>( & <mask>)"   ...  match/<name>: "<value>( & <mask>)" priority: <value> | ; One or more match fields and a priority value. The JSON components are described in the table below. |
+
+| KEY          | Format           | Description                                                                                                                                                                                                                                                                                                                                                                                      |
+|--------------|------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| match/<name> | String / Integer | Describes a match parameter for the rule. The match/<name> comes from the DESCRIPTION table. Matches may be exact or ternary values. See the Match Fields formatting section for details. Certain types may only appear as exact matches: SAI_ACL_TABLE_ATTR_FIELD_ACL_IP_TYPE/* SAI_ACL_TABLE_ATTR_FIELD_IN_PORT SAI_ACL_TABLE_ATTR_FIELD_OUT_PORT The value format depends on the match type.  |
+|     priority | Integer          | The priority of the ACL rule relative to other rules in the ACL table.                                                                                                                                                                                                                                                                                                                           |
+
+
+```
+127.0.0.1:6379> keys P4RT:ACL_ACL_INGRESS*
+ 1) "P4RT:ACL_ACL_INGRESS_TABLE:{\"match/dst_mac\":\"33:33:00:00:00:02&ff:ff:ff:ff:ff:ff\",\"match/icmpv6_type\":\"0x87&0xff\",\"match/ip_protocol\":\"0x3a&0xff\",\"match/is_ipv6\":\"0x1\",\"priority\":2070}"
```

Comment:
We can be aligned with the existing ACL_TABLE/ACL_RULE tables with unique key names, this would help for easy migration in the future with legacy tables.

Reply:
We chose json format so that it allows a wider choice of chars and also so that it doesnt conflict with existing chars in the key like ':','"' etc.

Reply:
I understand but this is definitely a deviation w.r.t APP_DB definition, it's better to be aligned with current APPL_DB tables so that we can easily migrate to common tables for SONiC regular and P4RT flows in the future.

Reply:
Sure, we will definitely look at moving these definitions closer to the regular SONiC table format as we get near to the unified tables approach.